tests/kepco.py

```diff
@@ -139,3 +141,44 @@ def test_GIVEN_psu_in_local_mode_WHEN_setpoint_is_sent_THEN_power_supply_put_int
         self.ca.process_pv(setpoint_pv)
 
         self._lewis.assert_that_emulator_value_is("remote_comms_enabled", True, cast=strtobool)
+
+    @skip_if_recsim("Calibration lookup does not work in recsim")
+    def test_GIVEN_calibration_of_none_WHEN_field_set_THEN_current_equal(self):
+        test_value = 100
+
+        with use_calibration_file(self.ca, "None.txt"):
+            self.ca.set_pv_value("FIELD:SP", test_value)
+            self.ca.assert_that_pv_is("CURRENT:SP", test_value)
+            self.ca.assert_that_pv_is("CURRENT:SP:RBV", test_value)
+            self.ca.assert_that_pv_is("FIELD:SP:RBV", test_value)
+
+    @skip_if_recsim("Calibration lookup does not work in recsim")
+    def test_GIVEN_calibration_of_double_amps_WHEN_field_set_THEN_current_double(self):
+        test_field = 100
+        expected_current = 50
+
+        with use_calibration_file(self.ca, "field_double_amps.txt"):
+            self.ca.set_pv_value("FIELD:SP", test_field)
+            self.ca.assert_that_pv_is("FIELD:SP:RBV", test_field)
+            self.ca.assert_that_pv_is("CURRENT:SP", expected_current)
+            self.ca.assert_that_pv_is("CURRENT:SP:RBV", expected_current)
+
+    @skip_if_recsim("Calibration lookup does not work in recsim")
+    def test_GIVEN_calibration_of_none_WHEN_current_read_THEN_field_equal(self):
+        test_current = 100
+        expected_field = 100
+
+        with use_calibration_file(self.ca, "None.txt"):
+            self._write_current(test_current)
+            self.ca.assert_that_pv_is("FIELD", expected_field)
+
+    @skip_if_recsim("Calibration lookup does not work in recsim")
+    def test_GIVEN_calibration_of_none_WHEN_current_read_THEN_field_equal(self):
```

Comment:
Perhaps a little bit over the top, but could this be parametrized with the file name and a tuple of the expected current and fields? or list of tuples, then you can test multiple values